TACC/Core-Styles
# fix: tup-639 move c-button font-weight bold to cms

## Overview

The `.c-button` font weight should be bold only on CMS.

And remove `font-weight: var(--medium)` because it is always overwritten.

## Related

- [TUP-639](https://jira.tacc.utexas.edu/browse/TUP-639)
- inspired by https://github.com/TACC/Core-Styles/pull/251#discussion_r1380667528
- required by https://github.com/TACC/Core-CMS/pull/740
- required by https://github.com/TACC/tup-ui/pull/364

## Changes

- **removed** font-weight medium of `c-button` not as a link
- **removed** font-weight bold of a `c-button` in a `c-card`

## Testing

1. Test this change on https://www.tacc.utexas.edu/education/k-12-students/high-school-camps/gencyber/#application.
    <sub>See https://github.com/wesleyboar/tup-ui-migration-to-core.</sub>
2. Verify "APPLICATION CLOSED" button is **bold**.
2. Verify "APPLICATION CLOSED" button has **no** instances of `font-weight: var(--medium)`.

## UI

| c-button | c-card |
| - | - |
| ![core-styles c-button](https://github.com/TACC/Core-Styles/assets/62723358/a384190d-6860-4fdb-8476-9fec86d38990) | ![core-styles c-card](https://github.com/TACC/Core-Styles/assets/62723358/5eeb459f-b224-4075-9b3b-30a31fdfd9c1) |
